File: week1-python/Getting_started_with_images/Saving_an_image.py
# Import libraries
import cv2
import numpy as np
from dataPath import DATA_PATH
import matplotlib.pyplot as plt
import matplotlib
matplotlib.rcParams['figure.figsize'] = (6.0, 6.0)
matplotlib.rcParams['image.cmap'] = 'gray'

imagePath = DATA_PATH + "/images/number_zero.jpg"
# Read image in Grayscale format
testImage = cv2.imread(imagePath,0)

#Write the Image to Disk
cv2.imwrite("../results/test.jpg",testImage)

File: week1-cpp/Getting_started_with_images/Saving_an_image.cpp
#include <iostream>
#include <opencv2/opencv.hpp>
#include <opencv2/core.hpp>
#include <opencv2/highgui.hpp>
#include "dataPath.hpp"

using namespace std;
using namespace cv;

int main(void)
{
        string imagePath = DATA_PATH + "/images/number_zero.jpg";

        // Read image in Grayscale format
        Mat testImage = imread(imagePath,0);

        testImage.at<uchar>(0,0)=200;

        Mat test_roi = testImage(Range(0,2),Range(0,4));
        testImage(Range(0,2),Range(0,4)).setTo(111);

	imwrite("../results/test.jpg",testImage);

        return 0;
}

